src/main/java/mate/service/AuthenticationServiceImpl.java

```diff
@@ -0,0 +1,26 @@
+package mate.service;
+
+import java.util.Optional;
+import mate.exception.AuthenticationException;
+import mate.lib.Inject;
+import mate.lib.Service;
+import mate.model.Driver;
+
+@Service
+public class AuthenticationServiceImpl implements AuthenticationService {
+    @Inject
+    private DriverService driverService;
+
+    @Override
+    public Driver login(String login, String password)
+            throws AuthenticationException {
+        Optional<Driver> driver = driverService.findByDriverLogin(login);
+        if (driver.isEmpty()) {
+            throw new AuthenticationException("Username or password was incorrect");
+        }
+        if (driver.get().getPassword().equals(password)) {
+            return driver.get();
```

Comment:
in try to avoid common mistakes written that if u can use only one if 
```suggestion
       Driver driver = driverService.findByLogin(login).orElseThrow(()
                -> new AuthenticationException("login or password is incorrect"));
        if (driver.getPassword().equals(password)) {
            return driver;
        }
```